foolbox/attacks/mi_fgsm.py

```diff
@@ -0,0 +1,108 @@
+from functools import partial
+from typing import Callable, Optional
+
+from foolbox.attacks.gradient_descent_base import normalize_lp_norms
+
+from .basic_iterative_method import (
+    Optimizer,
+    L1BasicIterativeAttack,
+    L2BasicIterativeAttack,
+    LinfBasicIterativeAttack,
+)
+import eagerpy as ep
+
+
+class GDMOptimizer(Optimizer):
+    # create GD optimizer with momentum
+    def __init__(
+        self,
+        x: ep.Tensor,
+        stepsize: float,
+        momentum: float = 1.0,
+        normalize_fn: Callable[[ep.Tensor], ep.Tensor] = lambda x: x.sign(),
+    ):
+        self.stepsize = stepsize
+        self.momentum = momentum
+        self.normalize = normalize_fn
+        self.accumulation_grad = ep.zeros_like(x)
+
+    def __call__(self, gradient: ep.Tensor) -> ep.Tensor:
+        self.accumulation_grad = self.momentum * self.accumulation_grad + gradient
+        return self.stepsize * self.normalize(self.accumulation_grad)
+
+
+class L1MomentumIterativeFastGradientMethod(L1BasicIterativeAttack):
+    def __init__(
+        self,
+        *,
+        momentum: float = 1.0,
+        rel_stepsize: float = 0.2,
+        abs_stepsize: Optional[float] = None,
+        steps: int = 10,
+        random_start: bool = False,
+    ):
+        self.momentum = momentum
+        super().__init__(
+            rel_stepsize=rel_stepsize,
+            abs_stepsize=abs_stepsize,
+            steps=steps,
+            random_start=random_start,
+        )
+
+    def get_optimizer(self, x: ep.Tensor, stepsize: float) -> Optimizer:
+        return GDMOptimizer(
+            x, stepsize, self.momentum, partial(normalize_lp_norms, p=1)
+        )
+
+
+class L2MomentumIterativeFastGradientMethod(L2BasicIterativeAttack):
+    def __init__(
+        self,
+        *,
+        momentum: float = 1.0,
+        rel_stepsize: float = 0.2,
+        abs_stepsize: Optional[float] = None,
+        steps: int = 10,
+        random_start: bool = False,
+    ):
+        self.momentum = momentum
+        super().__init__(
+            rel_stepsize=rel_stepsize,
+            abs_stepsize=abs_stepsize,
+            steps=steps,
+            random_start=random_start,
+        )
+
+    def get_optimizer(self, x: ep.Tensor, stepsize: float) -> Optimizer:
+        return GDMOptimizer(
+            x, stepsize, self.momentum, partial(normalize_lp_norms, p=2)
+        )
+
+
+class LinfMomentumIterativeFastGradientMethod(LinfBasicIterativeAttack):
+    """create I-FGSM with Momentum [#Dong18]
```

Comment:
Can you please fix the typo here (i.e., capitalize the first word)? Also, please use the same style of docstrings as for other attacks, i.e., don't start with "create I-FGSM with Momentum" but rather "Momentum Iterative Fast Gradient Sign Method (MI-FGSM)"?

Also, I think the URL in the reference is not correct.

Reply:
Also, it might be good to add a docstring to the other variants of the attack to make it clear what these attacks are.

Reply:
Thank you for your valuable comments. I have fixed these problems, and please let me know if there is any suggestion.

Reply:
Sure, thanks for updating the code quickly! I just noticed that it might be good to add a docstring to the other variants of the attack to make it clear what these attacks are.

Reply:
Thank you for your feedback and suggestions. I have implemented the changes.